private-identity-audit/src/proofs.rs

```diff
@@ -0,0 +1,156 @@
+//! Provides interactive ZKP for statements of muliple base.
+//!
+//! Statement: q=G^x
+//! ZKP(q, G):
+//!   1. P -> V: A = G^r
+//!   2. V -> P: c
+//!   3. P -> V: s = r + c*x
+//!   4. V: G^s == A * q^c
+//!
+//! Statement: q=H^y*F^z
+//! ZKP(q, G):
+//!   1. P -> V: A = H^r1 * F^r2
+//!   2. V -> P: c
+//!   3. P -> V: s1 = r1 + c*y, s2 = r2 + c*z
+//!   4. V: H^s1 * F^s2 == A * q^c
```

Comment:
We talked about it and decided to do the ZKP interactive since the protocol is already interactive anyway. So there is no need to add the complexity of Merlin transcript.